packages/pipeline/src/parsers/idex_orders/index.ts

```diff
@@ -0,0 +1,89 @@
+import { BigNumber } from '@0x/utils';
+import * as R from 'ramda';
+
+import { IdexOrder, IdexOrderbook, IdexOrderParam } from '../../data_sources/idex';
+import { TokenOrderbookSnapshot as TokenOrder } from '../../entities';
+import { OrderType } from '../../types';
+
+/**
+ * Marque function of this file.
+ * 1) Takes in orders from an orderbook,
+ * 2) Aggregates them by price point,
+ * 3) Parses them into entities which are then saved into the database.
+ * @param idexOrderbook raw orderbook that we pull from the Idex API.
+ * @param observedTimestamp Time at which the orders for the market were pulled.
+ * @param source The exchange where these orders are placed. In this case 'idex'.
+ */
+export function parseIdexOrders(idexOrderbook: IdexOrderbook, observedTimestamp: number, source: string): TokenOrder[] {
+    const aggregatedBids = aggregateOrders(idexOrderbook.bids);
+    // Any of the bid orders' params will work
+    const idexBidOrder = idexOrderbook.bids[0];
+    const parsedBids =
+        aggregatedBids.length > 0
+            ? aggregatedBids.map(order => parseIdexOrder(idexBidOrder.params, observedTimestamp, 'bid', source, order))
+            : [];
+
+    const aggregatedAsks = aggregateOrders(idexOrderbook.asks);
+    // Any of the ask orders' params will work
+    const idexAskOrder = idexOrderbook.asks[0];
+    const parsedAsks =
+        aggregatedAsks.length > 0
+            ? aggregatedAsks.map(order => parseIdexOrder(idexAskOrder.params, observedTimestamp, 'ask', source, order))
+            : [];
+    return parsedBids.concat(parsedAsks);
+}
+
+/**
+ * Aggregates orders by price point for consistency with other exchanges.
+ * The Idex API returns a breakdown of individual orders at each price point.
+ * Other exchanges only give total amount at each price point.
+ * Returns an array of <price, amount> tuples.
+ * @param idexOrders A list of Idex orders awaiting aggregation.
+ */
+export function aggregateOrders(idexOrders: IdexOrder[]): Array<[string, BigNumber]> {
```

Comment:
It looks like the only difference between this and the `aggregateOrders` function in the Paradex parser is the argument type. Can we generalize this and share the same function for both types of orders?

```ts
// Represents an order for an unspecified arbitrary pair. That is, it just has a price and amount.
interface UnspecifiedOrder {
    amount: BigNumber;
    price: BigNumber;
}

export function aggregateOrdersByPrice(orders: UnspecifiedOrder[]): Array<[string, BigNumber]> {
    // ...
}
```